example_scenes/advanced_tex_fonts.py

```diff
@@ -0,0 +1,691 @@
+
+# for the demo Scene
+from manim import (
+    BLACK,
+    WHITE,
+    UP,
+    UL,
+    ORIGIN,
+    DOWN,
+    LEFT,
+    RIGHT,
+    TextMobject,
+    ApplyMethod,
+    AnimationGroup,
+    VGroup,
+    FadeIn,
+    FadeOut,
+    Transform,
+    FadeOutAndShift,
+    FadeInFromDown,
+    FadeOutAndShiftDown,
+    Scene,
+)
+
+# Some font configurations specify the compiler as LaTeX, XeLaTeX, LuaLaTeX etc.
+# and also specify whether the code will compile to dvi or pdf format.
+# The goal is to read the font configuration from the TEX_FONT_CONFIGS dictionary
+# below and prepare a corresponding tex configration and template.
+
+# We then define FontMobject for convenience to easily add text to the demo scene
+# For example FontMobject("example", font="ecfaugieeg")
+
+# Then we define a demo scene that adds one line of every font.
+
+# WARNING: The demo scene will not compile unless all relevant fonts are
+# installed on your system. If a font (or LaTeX package) is missing, you will
+# get a Tex compilation error. You can then try to remebdy the error by installing
+# the missing font, or you can comment out the relevant entry in the TEX_FONT_CONFIGS
+# dictionary and try again without that font.
+
+
+class TexTemplateFromFontConfig(TexTemplateFromFile):
+    def __init__(self, font, **kwargs):
+        digest_config(self, kwargs)
+        self.font = font
+        if "compiler" in TEX_FONT_CONFIGS[font]:
+            tex_command = TEX_FONT_CONFIGS[font]["compiler"]
+        else:
+            tex_command = "latex"
+        if "output" in TEX_FONT_CONFIGS[font]:
+            tex_output = TEX_FONT_CONFIGS[font]["output"]
+        else:
+            tex_output = ".dvi"
+        self.tex_compiler = {
+            "command": tex_command,
+            "output_format": tex_output,
+        }
+        self.rebuild_cache()
+
+    def rebuild_cache(self):
+        self.body = self.GenerateTexTemplateForFont(self.font)
+
+    def GenerateTexTemplateForFont(self, font_name):
+        # TODO figure out how to do this with TexTemplate()
+        template = r"""
+\documentclass[preview]{standalone}
+
+\usepackage[english]{babel}
+\usepackage{amsmath}
+\usepackage{amssymb}
+
+"""
+        template += TEX_FONT_CONFIGS[font_name]["config"]
+        template += (
+            ""
+            if r"\begin{document}" in TEX_FONT_CONFIGS[font_name]["config"]
+            else r"""
+\begin{document}
+            """
+        )
+        template += r"""
+YourTextHere
+
+\end{document}
+"""
+        return template
+
+
+class FontMobject(TextMobject):
+    def __init__(self, *tex_strings, font=None, **kwargs):
+        if font is None:  # default to TextMobject
+            super().__init__(*tex_strings, **kwargs)
+        else:
+            super().__init__(
+                *tex_strings, template=TexTemplateFromFontConfig(font), **kwargs
+            )
+
+
+# All the font examples from http://jf.burnol.free.fr/showcase.html
+
+TEX_FONT_CONFIGS = {
```

Comment:
I suggest you write all these in a JSON file and then load it instead of embedding it. This makes this file more readable.

Reply:
It would be great if someone could instead make these into a library of TexTemplate objects. But tbh. I never understood how to properly use the new TexTemplate system.

Reply:
Oh I see, yeah I agree these would be better as a gallery. Or even something like `geometry.py` but for latex. Anyway, this is why we need more documentation :) I forget who implemented the template system - was it @PhilippImhof ?